app/src/main/java/com/example/meetup/LoginActivity.java

```diff
@@ -0,0 +1,148 @@
+    GoogleSignInClient mGoogleSignInClient;
+    public static final String TAG = "LoginActivity";
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        progressDialog = new ProgressDialog(LoginActivity.this);
+
+        username = findViewById(R.id.username);
+        password = findViewById(R.id.password);
+        login = findViewById(R.id.login);
+        navigateSignup = findViewById(R.id.navigatesignup);
+
+        login.setOnClickListener(v -> login(username.getText().toString(), password.getText().toString()));
+
+        navigateSignup.setOnClickListener(v -> {
+            startActivity(new Intent(LoginActivity.this, SignUpActivity.class));
+        });
+
+        setUpGoogleSignIn();
+        SignInButton googleSignIn = findViewById(R.id.sign_in_button);
+
+        Objects.requireNonNull(googleSignIn).setOnClickListener(v -> {
+            if (v.getId() == R.id.sign_in_button) {
+                signIn();
+            }
+        });
+    }
+
+    private void login(String username, String password) {
+        progressDialog.show();
+
+        ParseUser.logInInBackground(username, password, (parseUser, e) -> {
+            progressDialog.dismiss();
+            if (parseUser != null) {
+                showAlert("Successful Login", "Welcome back " + username + " !");
+            } else {
+                ParseUser.logOut();
+                Toast.makeText(LoginActivity.this, e.getMessage(), Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+
+    private void showAlert(String title,String message){
+        AlertDialog.Builder builder = new AlertDialog.Builder(LoginActivity.this)
+                .setTitle(title)
+                .setMessage(message)
+                .setPositiveButton("OK", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.cancel();
+                        Intent intent = new Intent(LoginActivity.this, LogoutActivity.class);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
+                        startActivity(intent);
+                    }
+                });
+        AlertDialog ok = builder.create();
+        ok.show();
+    }
+
+    private void setUpGoogleSignIn() {
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail()
+                .build();
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
+
+        GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(this);
+        if (account != null) {
+            goMainActivity();
+        }
+    }
+
+    private void signIn() {
```

Comment:
The naming of this method may be a little bit confusing. I guess `googleSignIn` or `signInWithGoogle` will be better names. What do you think?

Reply:
renaming it to googleSignIn